### PR TITLE
[circle-mpqsolver] Introduce getSampleInput

### PR DESCRIPTION
**Pull request by stamalakhov**
This commit introduces getSampleInput for using in tests.

Draft: https://github.com/Samsung/ONE/pull/11849
Related: https://github.com/Samsung/ONE/issues/11374

ONE-DCO-1.0-Signed-off-by: s.malakhov <s.malakhov@partner.samsung.com>

### CODE REVIEW

File: compiler/circle-mpqsolver/src/core/TestHelper.test.cpp [hunk 1]
```diff
@@ -208,5 +208,36 @@ bool isFileExists(const std::string &path)
...
+    size_t size = data.data().size() / sizeof(float);
+    auto floats = reinterpret_cast<float *>(data.data().data());
+    for (uint32_t idx = 0; idx < size; idx++)
+    {
+      floats[idx] = 0.f; // or any other value
+    }
+  }
+
+private:
+  uint32_t _num_of_samples;
```
Comment (seanshpark) [inline compiler/circle-mpqsolver/src/core/TestHelper.test.cpp:232]: I don't know much about `DataProvider`,
if `SampleDataInput` only provides single sample, then `numInputs` uses `1` not `_num_of_samples` the `_num_of_samples` seems useless.

File: compiler/circle-mpqsolver/src/core/TestHelper.test.cpp [hunk 2]
```diff
@@ -208,5 +208,36 @@ bool isFileExists(const std::string &path)
 }
 
 } // namespace io_utils
+
+namespace data_utils
+{
+
+class SampleDataInput final : public core::DataProvider
```
Comment (seanshpark) [inline compiler/circle-mpqsolver/src/core/TestHelper.test.cpp:215]: optional) not understanding much about this but how about `SingleDataProvider` ?

File: compiler/circle-mpqsolver/src/core/TestHelper.test.cpp [hunk 3]
```diff
@@ -208,5 +208,36 @@ bool isFileExists(const std::string &path)
...
+{
+public:
+  SampleDataInput() : _num_of_samples(1) {}
+  size_t numSamples() const override { return _num_of_samples; }
+  uint32_t numInputs(uint32_t sample) const override { return 1; }
```
Comment (seanshpark) [inline compiler/circle-mpqsolver/src/core/TestHelper.test.cpp:220]:
```suggestion
  uint32_t numInputs(uint32_t) const override { return 1; }
```

File: compiler/circle-mpqsolver/src/core/TestHelper.test.cpp [hunk 4]
```diff
@@ -208,5 +208,36 @@ bool isFileExists(const std::string &path)
...
+  void getSampleInput(uint32_t sample, uint32_t input, core::InputData &data) const
```
Comment (seanshpark) [inline compiler/circle-mpqsolver/src/core/TestHelper.test.cpp:221]:
```suggestion
  void getSampleInput(uint32_t, uint32_t, core::InputData &data) const
```

File: compiler/circle-mpqsolver/src/core/TestHelper.test.cpp [hunk 5]
```diff
@@ -208,5 +208,36 @@ bool isFileExists(const std::string &path)
...
+  {
+    size_t size = data.data().size() / sizeof(float);
```
Comment (seanshpark) [inline compiler/circle-mpqsolver/src/core/TestHelper.test.cpp:223]: Q) just curious and my be stupid question, if we can use number of elements in `core::InputData` are used, why do we pass `sample` and `input`?

File: compiler/circle-mpqsolver/src/core/TestHelper.test.cpp (hunk 1 above)
Comment (stamalakhov) [reply compiler/circle-mpqsolver/src/core/TestHelper.test.cpp:232]: Ahh. Yes. Sorry. You're right. I'll remove it.

File: compiler/circle-mpqsolver/src/core/TestHelper.test.cpp (hunk 2 above)
Comment (stamalakhov) [reply compiler/circle-mpqsolver/src/core/TestHelper.test.cpp:215]: > optional) not understanding much about this but how about `SingleDataProvider` ?

Yes. Looks better. Thank you.

File: compiler/circle-mpqsolver/src/core/TestHelper.test.cpp (hunk 5 above)
Comment (stamalakhov) [reply compiler/circle-mpqsolver/src/core/TestHelper.test.cpp:223]: > Q) just curious and my be stupid question, if we can use number of elements in `core::InputData` are used, why do we pass `sample` and `input`?

@seanshpark
`Sample` stands for specific `data_point`, this `data_point` in its turn can have any number of tensors, so it's two-dimensional, that is why we need two parameters to query specific data. `size` it's just a number of elements in `data`. These parameters are unrelated. So we need them all. Quantity of inputs for specific `sample` can be queried by `numInputs`.

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Review by jinevening — APPROVED** [review]
LGTM

**Pull request merged**